### DISCUSSION TITLE
Incluir nova forma para percorrer itens

### DISCUSSION THREAD

**Discussion by samuelrvg** [DesignLiquido/delegua#342] [Ideias]
Seguindo a ideia do usuário [JohnPetros](https://github.com/JohnPetros) no canal do [Discord](https://discord.gg/4tBxWSSbdV), poderíamos implementar novas formas de percorrer uma lista, assim como temos no `JavaScript`.

No `js` temos `for .. in` e `for .. of`

Algumas sugestões:

- `para (var item de meu_vetor) { ... }` @samuelrvg 
- `para cada elemento em meu_vetor { ... }` @leonelsanchesdasilva

**Comment by leonelsanchesdasilva** [comment]
Estendo minha proposta para aceitar duas formas:

```
para cada elemento em meu_vetor {
    escreva(elemento)
}
```

Ou 

```
para cada elemento de meu_vetor {
    escreva(elemento)
}
```

**Comment by leonelsanchesdasilva** [comment]
Para iterar chave e valor, eu faria assim:

```
para cada chave, valor em meu_vetor {
    escreva(chave, ":", valor)
}
```

Ou

```
para cada chave, valor de meu_vetor {
    escreva(chave, ":", valor)
}
```

**Reply by samuelrvg** [reply]
Eu já penso em algo mais parecido com `js` e outras linguagens como:

```
escreva(objeto.valor)
```

**Reply by leonelsanchesdasilva** [reply]
@samuelrvg JS tem essa forma de uso. É mais ou menos assim:

```
for (const [key, value] of Object.entries(test)) {
  console.log(key, value);
}
```

Mas não sou fã dessa forma. Gosto mais de uma coisa perto do Python.

**Comment by leonelsanchesdasilva** [comment]
`para cada` foi implementado [na versão 0.19.0](https://github.com/DesignLiquido/delegua/releases/tag/0.19.0).
